utils/include/utils/resource_location.h

```diff
@@ -57,6 +57,11 @@ class resource_location
     resource_location& replace_filename(std::string_view other);
     static resource_location replace_filename(resource_location rl, std::string_view other);
 
+    std::string filename() const;
+    resource_location parent() const;
+
+    std::string get_local_if_available() const;
```

Comment:
Hm, based on the function name I would expect it to return either a local path, empty string or `nullopt`.

I suggest one of the following:
1. returning an empty string or `nullopt` when local is not available
2. renaming to something like "get_local_or_non_local_uri", "get_local_or_uri"